File: src/main/java/com/wlyscraft/slimefunitemexpansion/machines/AlloyFurnace.java
package com.wlyscraft.slimefunitemexpansion.machines;

import com.wlyscraft.slimefunitemexpansion.items.Items;
import com.wlyscraft.slimefunitemexpansion.SlimefunItemExpansion;
import io.github.thebusybiscuit.slimefun4.core.attributes.RecipeDisplayItem;
import io.github.thebusybiscuit.slimefun4.implementation.SlimefunItems;
import me.mrCookieSlime.Slimefun.Lists.RecipeType;
import me.mrCookieSlime.Slimefun.Objects.Category;
import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.AContainer;
import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
import me.mrCookieSlime.Slimefun.cscorelib2.item.CustomItem;
import org.bukkit.Material;
import org.bukkit.NamespacedKey;
import org.bukkit.inventory.ItemStack;

import javax.annotation.Nonnull;

public class AlloyFurnace extends AContainer implements RecipeDisplayItem {
    public AlloyFurnace(Category category, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
        super(category, item, recipeType, recipe);
    }

    @Override
    public ItemStack getProgressBar() {
        return new ItemStack(Material.FLINT_AND_STEEL);
    }


    @Nonnull
    @Override
    public String getMachineIdentifier() {
        return "IE_ALLOY_FURNACE";
    }


    @Override
    public void registerDefaultRecipes() {
        this.registerRecipe(1, new ItemStack[]{new CustomItem(Material.KNOWLEDGE_BOOK, "§b配方表", "§b1圆石齿轮+1铁锭=1铁齿轮", "§b1沙子+1金粉=1金锭(原版)", "§b1锇锭+1荧石粉=1强化荧石锭", "§b1黑曜石+1硬化玻璃=1强化黑曜石", "§b1强化荧石锭+1强化黑曜石=1锇荧黑曜合金锭")}, new ItemStack[]{new CustomItem(Material.KNOWLEDGE_BOOK, "§b配方表", "§b1圆石齿轮+1铁锭=1铁齿轮", "§b1沙子+1金粉=1金锭(原版)", "§b1锇锭+1荧石粉=1强化荧石锭", "§b1黑曜石+1硬化玻璃=1强化黑曜石", "§b1强化荧石锭+1强化黑曜石=1锇荧黑曜合金锭")});
        this.registerRecipe(8, new ItemStack[]{Items.COBBLESTONE_GEAR, new ItemStack(Material.IRON_INGOT)}, new ItemStack[]{Items.IRON_GEAR});
        this.registerRecipe(8, new ItemStack[]{new ItemStack(Material.SAND), SlimefunItems.GOLD_DUST}, new ItemStack[]{new ItemStack(Material.GOLD_INGOT)});
        this.registerRecipe(10, new ItemStack[]{Items.OSMIUM_INGOT, new ItemStack(Material.GLOWSTONE_DUST)}, new ItemStack[]{Items.ENHANCED_GLOWSTONE});
        this.registerRecipe(12, new ItemStack[]{new ItemStack(Material.OBSIDIAN), SlimefunItems.HARDENED_GLASS}, new ItemStack[]{Items.ENHANCED_OBSIDIAN});
        this.registerRecipe(16, new ItemStack[]{Items.ENHANCED_GLOWSTONE, Items.ENHANCED_OBSIDIAN}, new ItemStack[]{Items.OSGLOGLAS_INGOT});
    }

    public static RecipeType recipeType() {
        return new RecipeType(new NamespacedKey(SlimefunItemExpansion.getInstance(), "alloy_furnace"), Items.ALLOY_FURNACE);
    }

}

File: src/main/kotlin/io/github/itsflicker/slimefunitemexpansion/items/machines/AlloyFurnace.kt
package io.github.itsflicker.slimefunitemexpansion.items.machines

import io.github.itsflicker.slimefunitemexpansion.items.IEItems
import io.github.itsflicker.slimefunitemexpansion.util.toKey
import io.github.thebusybiscuit.slimefun4.core.attributes.RecipeDisplayItem
import io.github.thebusybiscuit.slimefun4.implementation.SlimefunItems
import me.mrCookieSlime.Slimefun.Lists.RecipeType
import me.mrCookieSlime.Slimefun.Objects.Category
import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.AContainer
import me.mrCookieSlime.Slimefun.api.SlimefunItemStack
import me.mrCookieSlime.Slimefun.cscorelib2.item.CustomItem
import org.bukkit.Material
import org.bukkit.inventory.ItemStack

class AlloyFurnace(category: Category, item: SlimefunItemStack, recipeType: RecipeType, recipe: Array<ItemStack?>) :
    AContainer(category, item, recipeType, recipe), RecipeDisplayItem {

    override fun getProgressBar(): ItemStack {
        return ItemStack(Material.FLINT_AND_STEEL)
    }

    override fun getMachineIdentifier(): String {
        return "IE_ALLOY_FURNACE"
    }

    public override fun registerDefaultRecipes() {
        this.registerRecipe(
            1,
            arrayOf<ItemStack>(
                CustomItem(
                    Material.KNOWLEDGE_BOOK,
                    "§b配方表",
                    "§b1圆石齿轮+1铁锭=1铁齿轮",
                    "§b1沙子+1金粉=1金锭(原版)",
                    "§b1锇锭+1荧石粉=1强化荧石锭",
                    "§b1黑曜石+1压缩经验方块=1强化黑曜石",
                    "§b1强化荧石锭+1强化黑曜石=1锇荧黑曜合金锭"
                )
            ),
            arrayOf(
                ItemStack(Material.AIR)
            )
        )
        this.registerRecipe(
            8,
            arrayOf(IEItems.COBBLESTONE_GEAR, ItemStack(Material.IRON_INGOT)),
            arrayOf<ItemStack>(IEItems.IRON_GEAR)
        )
        this.registerRecipe(
            8,
            arrayOf(ItemStack(Material.SAND), SlimefunItems.GOLD_DUST),
            arrayOf(ItemStack(Material.GOLD_INGOT))
        )
        this.registerRecipe(
            10,
            arrayOf(IEItems.OSMIUM_INGOT, ItemStack(Material.GLOWSTONE_DUST)),
            arrayOf<ItemStack>(IEItems.ENHANCED_GLOWSTONE)
        )
        this.registerRecipe(
            12,
            arrayOf(ItemStack(Material.OBSIDIAN), SlimefunItems.HARDENED_GLASS),
            arrayOf<ItemStack>(IEItems.ENHANCED_OBSIDIAN)
        )
        this.registerRecipe(
            16,
            arrayOf<ItemStack>(IEItems.ENHANCED_GLOWSTONE, IEItems.ENHANCED_OBSIDIAN),
            arrayOf<ItemStack>(IEItems.OSGLOGLAS_INGOT)
        )
    }

    companion object {

        val recipeType by lazy {
            RecipeType("alloy_furnace".toKey(), IEItems.ALLOY_FURNACE)
        }
    }
}